devmand/gateway/src/devmand/devices/cli/StructuredUbntDevice.cpp

```diff
@@ -303,25 +303,29 @@ StructuredUbntDevice::StructuredUbntDevice(
     const shared_ptr<CliCache> _cmdCache)
     : Device(application, id_, true), channel(_channel), cmdCache(_cmdCache) {}
 
-
-void StructuredUbntDevice::setConfig(const dynamic & config) {
-  const string &json = folly::toJson(config);
-  auto &bundle = mreg->getBundle(Model::OPENCONFIG_0_1_6);
-  const shared_ptr<OpenconfigInterfaces> &ydkModel = make_shared<OpenconfigInterfaces>();
+void StructuredUbntDevice::setConfig(const dynamic& config) {
+  const string& json = folly::toJson(config);
+  auto& bundle = mreg->getBundle(Model::OPENCONFIG_0_1_6);
+  const shared_ptr<OpenconfigInterfaces>& ydkModel =
+      make_shared<OpenconfigInterfaces>();
   const shared_ptr<Entity> decodedIfcEntity = bundle.decode(json, ydkModel);
   MLOG(MDEBUG) << decodedIfcEntity->get_segment_path();
 
-    for (shared_ptr<Entity> entity : ydkModel->interface.entities()) {
-       shared_ptr<OpenconfigInterface> iface = std::static_pointer_cast<OpenconfigInterface >(entity);
-       shared_ptr<OpenconfigConfig> openConfig = iface->config;
-       if(openConfig->type.get() != ietf::iana_if_type::EthernetCsmacd().to_string()) {
-           continue;
-       }
-       string enabled = openConfig->enabled.get(); //TODO YLeaf does not support bool
-       string name = openConfig->name;
-       channel->execute(createInterfaceCommand(name, enabled == "true" ? true : false));
-       return;
+  for (shared_ptr<Entity> entity : ydkModel->interface.entities()) {
```

Comment:
why casting to entities and then back to openconfigInterface ?

Reply:
no reason, I will change it to direct cast